src/AppInstallerCLICore/Workflows/DependenciesFlow.cpp

```diff
@@ -133,6 +135,109 @@ namespace AppInstaller::CLI::Workflow
         }
     }
 
+    void EnableWindowsFeaturesDependencies(Execution::Context& context)
+    {
+        if (!Settings::ExperimentalFeature::IsEnabled(Settings::ExperimentalFeature::Feature::WindowsFeature))
+        {
+            return;
+        }
+
+        const auto& rootDependencies = context.Get<Execution::Data::Installer>()->Dependencies;
+
+        if (rootDependencies.Empty())
+        {
+            return;
+        }
+
+        if (rootDependencies.HasAnyOf(DependencyType::WindowsFeature))
+        {
+            context << Workflow::EnsureRunningAsAdmin;
+            if (context.IsTerminated())
+            {
+                return;
+            }
+
+            std::vector<std::string> invalidFeatures;
+
+            rootDependencies.ApplyToType(DependencyType::WindowsFeature, [&invalidFeatures](Dependency dependency)
+                {
+                    const auto& name = dependency.Id();
+                    WindowsFeature::WindowsFeature windowsFeature{ name };
+                    if (!windowsFeature.DoesExist())
+                    {
+                        invalidFeatures.emplace_back(name);

```

Comment:
Instead of using memory to create a vector, why not have a string like: `<value>Windows Feature not found: {0}</value>` and list the features out as they are found to be invalid, and make invalidFeatures a bool; Then, later you can show the message `<value>One or more dependencies were not found; proceeding due to --ignore-missing-dependencies.</value>`  (or the equivalent for not proceeding) which could be used for any dependency type.

Then you wouldn't need the bool for whether or not to terminate, and avoid the iterator for invalidFeatures

Reply:
no longer create vector to aggregate missing dependencies, they will be reported at the end after each windows feature is processed